docker-app/qfieldcloud/core/utils.py

```diff
@@ -109,13 +109,22 @@ def get_s3_session() -> boto3.Session:
 
 
 def get_s3_bucket() -> mypy_boto3_s3.service_resource.Bucket:
-    """Get a new S3 Bucket instance using Django settings"""
+    """
+    Get a new S3 Bucket instance using Django settings.
+    """
 
-    session = get_s3_session()
+    bucket_name = settings.STORAGE_BUCKET_NAME
+
+    assert bucket_name, "Expected `bucket_name` to be non-empty string!"
 
-    # Get the bucket objects
+    session = get_s3_session()
     s3 = session.resource("s3", endpoint_url=settings.STORAGE_ENDPOINT_URL)
-    return s3.Bucket(settings.STORAGE_BUCKET_NAME)
+
+    # Ensure the bucket exists
+    s3.meta.client.head_bucket(Bucket=bucket_name)
+
+    # Get the bucket resource
+    return s3.Bucket(bucket_name)
```

Comment:
What I am still wondering is what will happen if we skip the line with `head_bucket` and the bucket does not exist on `s3.Bucket`. Aren't we going to get the same error?

Reply:
What do you mean by "skip the line"?

Reply:
If we don't have it at all, sorry.

I think the error would be the same with or without that line (124) in case the bucket does not exist?

Reply:
On an nonexistent Bucket, line 124 will raise `NoSuchBucket`. I cannot guarantee the same exception would be raised on line 127 if the Bucket doesn't exist.

Reply:
Not sure if you "cannot guarantee" because you observed different behaviour when you tested or because you didn't test?

Reply:
In my tests, if you remove the line 124 and passes a non-empty string that references no bucket to `s3.Bucket(...)`, you get no error from `get_s3_bucket()`. Line 124 is there to prevent that from happening.

Reply:
Ok, let's merge it, but we need to monitor the average response time if it increases due to the extra HEAD call to the Object storage.